src/args.rs

```diff
@@ -56,3 +68,37 @@ pub struct Args {
     #[arg(long)]
     pub license: bool,
 }
+
+/// Converts the arguments to a string that can be passed to a sub-make invocation.
+pub fn args_to_submake_str() -> String {
+    // Rudimetary MAKEFLAGS parsing, the '-j' flag handling is not implemented yet.
+    // TODO: This should probably be a `Result` instead of a `panic!`.
+    // NOTE: Maybe change the way this is done to a pure IPC solution? when sub-make is used?
+    let args = std::env::args()
+        .collect::<Vec<_>>()
+        .iter()
+        .map(|arg| {
+            let mut arg_mod = arg.clone();
+
+            // If the argument contains a space, we need to quote it.
+            if arg_mod.contains(' ') {
+                arg_mod = format!("\"{}\"", arg_mod);
+            }
+
+            if arg_mod.starts_with("--") {
+                arg_mod = arg_mod[2..].to_string();
+            } else if arg_mod.starts_with('-') {
+                arg_mod = arg_mod[1..].to_string();
+            }
+
+            // remove special cases
+            if SUBMAKE_FORBIDDEN_FLAGS.contains(&arg_mod.as_str()) {
+                arg_mod = "".to_string();
+            }
```

Comment:
Not 100% sure, but I think this would fail for an argument like `-abc`, where it's 3 single flags passed as a single arg. I know `clap` handles that, so it might be better to add this as a method under `impl Args` and use the struct information to construct args to pass to any sub-make.

If I'm wrong about this not working for coalesced flags, then feel free to ignore this comment.

Reply:
I will get around to this one tomorrow, should we add the args to the context?

Reply:
I don't think context should contain the args, I just think the helper should use the args struct that is attached to the Makefile.